library/core/src/hint.rs

```diff
@@ -786,12 +786,42 @@ pub fn select_unpredictable<T>(condition: bool, true_val: T, false_val: T) -> T
     // Change this to use ManuallyDrop instead.
     let mut true_val = MaybeUninit::new(true_val);
     let mut false_val = MaybeUninit::new(false_val);
+
+    struct DropOnPanic<T> {
+        inner: *mut MaybeUninit<T>,
```

Comment:
References should be fine here, right?

```suggestion
        inner: &'a mut MaybeUninit<T>,
```

Reply:
We have the same two pointers in both arguments to the first `select_unpredictable`, each one duplicate. So no.

Reply:
Oh right, fair enough!

Reply:
Do you think it's simpler code to dereference the pointer as it gets constructed into a `Guard`? I thought since we're calling `MaybeUninit`'s proxy to `drop_in_place` it would just add one additional location of unsafe precondition,

Reply:
Another option would be to avoid the `MaybeUninit` entirely in the type and use just raw pointers with direct `drop_in_place` calls. The `assume_init_drop` is noise with an additional precondition that must be fulfilled anyways. And if the comment of moving to `ManuallyDrop` is eventually followed-up on those attributes would remain untouched.

Reply:
> Another option would be to avoid the `MaybeUninit` entirely in the type and use just raw pointers with direct `drop_in_place` calls. The `assume_init_drop` is noise with an additional precondition that must be fulfilled anyways. And if the comment of moving to `ManuallyDrop` is eventually followed-up on those attributes would remain untouched.

Personally, I think this might be the cleanest option.